include/circt/Dialect/MSFT/MSFTPDOps.td

```diff
@@ -65,8 +65,22 @@ def PDPhysRegionOp : MSFTOp<"pd.physregion",
   }];
 }
 
+def InstanceHierarchyOp : MSFTOp<"instance.hierarchy",
```

Comment:
Nit: please add a little description for the docs to go alongside instance.dynamic. Are there any restrictions? For example, what can topModuleRef be? Could this point to the middle of the dynamic instance hierarchy? I assume so, since there can be multiple in one ModuleOp, but it would be good to add a little description of how to use it.